langchain/agents/tools.py

```diff
@@ -1,26 +1,19 @@
 """Interface for tools."""
-import asyncio
-from dataclasses import dataclass
+
 from inspect import signature
-from typing import Any, Awaitable, Callable, Optional, Union
+from typing import Any, Callable, Union
+
+from langchain.tools.tool import Tool
 
 
-@dataclass
-class Tool:
-    """Interface for tools."""
+class DynamicTool(Tool):
```

Comment:
If we want to still have the wrapper instantiation, this would be one way (naming tbd) we could keep it while still having the more structured def elsewhere